[CI] IndexSnapshotsServiceIT testGetShardSnapshotReturnsTheLatestSuccessfulSnapshot failing

The specific seed reproduces for me locally on master

**Build scan:**
https://gradle-enterprise.elastic.co/s/7gvsf2tvofpzu/tests/:server:internalClusterTest/org.elasticsearch.repositories.IndexSnapshotsServiceIT/testGetShardSnapshotReturnsTheLatestSuccessfulSnapshot

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.repositories.IndexSnapshotsServiceIT.testGetShardSnapshotReturnsTheLatestSuccessfulSnapshot" -Dtests.seed=88FB747CD5E68E3B -Dtests.locale=tr-TR -Dtests.timezone=Europe/Tallinn -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.repositories.IndexSnapshotsServiceIT&tests.test=testGetShardSnapshotReturnsTheLatestSuccessfulSnapshot

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: "okMS9jAVQiqFbaD2KDW2vA-_na_-1-2-1"
     but: was "okMS9jAVQiqFbaD2KDW2vA-_na_-1-1-1"

  at __randomizedtesting.SeedInfo.seed([88FB747CD5E68E3B:78BE1FA83EDBC669]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.repositories.IndexSnapshotsServiceIT.testGetShardSnapshotReturnsTheLatestSuccessfulSnapshot(IndexSnapshotsServiceIT.java:157)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/repositories/IndexSnapshotsServiceIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/repositories/IndexSnapshotsServiceIT.java
@@ -124,6 +124,8 @@ public class IndexSnapshotsServiceIT extends AbstractSnapshotIntegTestCase {
         List<String> indices = List.of(indexName, indexName2);
         createIndex(indexName, indexName2);
         SnapshotInfo lastSnapshot = null;
+        String expectedIndexMetadataId = null;
+
         int numSnapshots = randomIntBetween(5, 25);
         for (int i = 0; i < numSnapshots; i++) {
             if (randomBoolean()) {
@@ -134,6 +136,9 @@ public class IndexSnapshotsServiceIT extends AbstractSnapshotIntegTestCase {
             final SnapshotInfo snapshotInfo = createSnapshot(repoName, String.format(Locale.ROOT, "snap-%03d", i), snapshotIndices);
             if (snapshotInfo.indices().contains(indexName)) {
                 lastSnapshot = snapshotInfo;
+                ClusterStateResponse clusterStateResponse = admin().cluster().prepareState().execute().actionGet();
+                IndexMetadata indexMetadata = clusterStateResponse.getState().metadata().index(indexName);
+                expectedIndexMetadataId = IndexMetaDataGenerations.buildUniqueIdentifier(indexMetadata);
             }
         }
 
@@ -151,10 +156,7 @@ public class IndexSnapshotsServiceIT extends AbstractSnapshotIntegTestCase {
 
             final ShardSnapshotInfo shardSnapshotInfo = indexShardSnapshotInfoOpt.get();
 
-            final ClusterStateResponse clusterStateResponse = admin().cluster().prepareState().execute().actionGet();
-            final IndexMetadata indexMetadata = clusterStateResponse.getState().metadata().index(indexName);
-            final String indexMetadataId = IndexMetaDataGenerations.buildUniqueIdentifier(indexMetadata);
-            assertThat(shardSnapshotInfo.getIndexMetadataIdentifier(), equalTo(indexMetadataId));
+            assertThat(shardSnapshotInfo.getIndexMetadataIdentifier(), equalTo(expectedIndexMetadataId));
 
             final Snapshot snapshot = shardSnapshotInfo.getSnapshot();
             assertThat(snapshot, equalTo(lastSnapshot.snapshot()));

